### ISSUE TITLE
Error on update_tbdb

### ISSUE THREAD

**Issue by pmenzel** [jodyphelan/TBProfiler#188]
Hi Jody,

I installed TB-Profiler 4.0.2 from bioconda (btw, the version number in `__init__.py` is behind) and run `tb-profiler update_tbdb`, which produced this error:
```
Running command:
set -u pipefail; git clone https://github.com/jodyphelan/tbdb.git

Running command:
set -u pipefail; git pull

Running command:
set -u pipefail; tb-profiler create_db 
Traceback (most recent call last):
  File "/opt/conda/miniconda3/envs/tb42/bin/tb-profiler", line 594, in <module>
    args.func(args)
  File "/opt/conda/miniconda3/envs/tb42/bin/tb-profiler", line 231, in main_update_tbdb
    pp.run_cmd("tb-profiler create_db %s" % tmp)
  File "/opt/conda/miniconda3/envs/tb42/lib/python3.9/site-packages/pathogenprofiler/utils.py", line 255, in run_cmd
    raise ValueError("Command Failed:\n%s\nstderr:\n%s" % (cmd,stderr.decode()))
ValueError: Command Failed:
set -u pipefail; tb-profiler create_db 
stderr:

Running command:
set -u pipefail; snpEff ann Mycobacterium_tuberculosis_h37rv 16b3d18f-c790-4317-ab8f-c0c12510301e
Traceback (most recent call last):
  File "/opt/conda/miniconda3/envs/tb42/bin/tb-profiler", line 594, in <module>
    args.func(args)
  File "/opt/conda/miniconda3/envs/tb42/bin/tb-profiler", line 237, in main_create_db
    tbp.create_db(args)
  File "/opt/conda/miniconda3/envs/tb42/lib/python3.9/site-packages/tbprofiler/db.py", line 405, in create_db
    mut = mutation_lookup[(row["Gene"],row["Mutation"])]
KeyError: ('ddn', 'c.51_101delTATCAAGTGGATGTCACGGATTAATACCTGGATGTACCGCCGCAACGACGG')


################################# ERROR #######################################

This run has failed. Please check all arguments and make sure all input files
exist. If no solution is found, please open up an issue at
https://github.com/jodyphelan/TBProfiler/issues/new and paste or attach the
contents of the error log (tbdb.errlog)

###############################################################################

Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/opt/conda/miniconda3/envs/tb42/bin/tb-profiler", line 30, in cleanup
    with open(outfile, "w") as O:
UnboundLocalError: local variable 'outfile' referenced before assignment
```

It looks like, the problem is parsing this line in `tbdb.csv`:
```
ddn,c.51_101delTATCAAGTGGATGTCACGGATTAATACCTGGATGTACCGCCGCAACGACGG,delamanid,resistance,,PMID:29523322
```
It's the first line with a deletion.

Looking at `db.py`, the code for reading the deletions looks copy-pasted for both regexes:

https://github.com/jodyphelan/TBProfiler/blob/73f24612a3eb0d5cb64d83681b23c05691d60d53/tbprofiler/db.py#L135

https://github.com/jodyphelan/TBProfiler/blob/73f24612a3eb0d5cb64d83681b23c05691d60d53/tbprofiler/db.py#L152

Maybe that's the cause of the issue?

**Comment by jodyphelan** [comment]
Thanks @pmenzel I'll take a look!

**Comment by jodyphelan** [comment]
Hi @pmenzel 

I can't seem to recreate the error in my environment:
```
(test) jody@jody-Precision-Tower-3620:~/temp$ tb-profiler update_tbdb

Running command:
set -u pipefail; git clone https://github.com/jodyphelan/tbdb.git

Running command:
set -u pipefail; git pull

Running command:
set -u pipefail; tb-profiler create_db 

Running command:
set -u pipefail; tb-profiler load_library tbdb
Sucessfully updated TBDB
```

Which OS are you running this on?
Would it be possible for you to check if snpEff has been installed properly? (this might be connected to the issue)
If you have a VCF you can run `snpEff  ann Mycobacterium_tuberculosis_h37rv <test.vcf>`

**Comment by pmenzel** [comment]
I tested snpEff and it looks like that the problem is because snpEff fails to download the database from Sourceforge.
This is because the server is behind a proxy server and by default, the java runtime does not use any proxy servers, even if they are configured in the Linux environment.

One can tell java to do so by using `java -Djava.net.useSystemProxies=true`.  
SnpEff passes additional command line parameters on to java, so this command works:
```
snpEff -Djava.net.useSystemProxies=true ann Mycobacterium_tuberculosis_h37rv file.vcf
```

Now TB-profiler would need to call snpEff like this too.  
Could you change the snpEff call inside tb-profiler appropriatly?
I assume it should have no impact on systems without a proxy..

**Comment by jodyphelan** [comment]
Thanks for investigating. I am going to upgrade snpEff to v5 as I think this might help with the proxy issue (https://github.com/pcingola/SnpEff/issues/287). But in any case I will add `-Djava.net.useSystemProxies=true` and test this out.

**Comment by jodyphelan** [comment]
Does the pipeline work as normal after you have sucessfully downloaded the Mycobacterium_tuberculosis_h37rv database?

**Comment by pmenzel** [comment]
`tb-profiler update_tbdb` completed successfully now. I guess, after running the modified `snpEff ann` command, the database was downloaded to right place.

**Comment by jodyphelan** [comment]
Thanks for letting me know. I'll put in a fix for that asap.

**Comment by pmenzel** [comment]
Thanks, Jody!
If setting the proxy option works for you, maybe you could bump the minor version? Then it might find its way into conda soon.

The versioning of snpEff looks confusing. The latest tag in [pcingola/SnpEff ](https://github.com/pcingola/SnpEff) is v4.3t from 2017 and there are no releases.. Is v5 just the latest on the master branch?

**Comment by jodyphelan** [comment]
Yeah it looks like the v5 releases are not on gihutb. They are bein sourced from a different location in bioconda:

https://github.com/bioconda/bioconda-recipes/blob/1e167c3fd751fdde5a05c6ff7250fd137a194f10/recipes/snpeff/meta.yaml#L20

**Comment by jodyphelan** [comment]
Thanks for all your help, this should be fixed in release [v4.0.3](https://github.com/jodyphelan/TBProfiler/releases/tag/v4.0.3) and is available on [bioconda](https://bioconda.github.io/recipes/tb-profiler/README.html).

**Comment by pmenzel** [comment]
Great, I will test it tomorrow!
